### PR TITLE
Cleans up some spawn()s in magnetcode

### PR DESCRIPTION
**Pull request by ghost**
:cl: More Robust Than You
code: Cleans up some spawn()s in magnetcode
/:cl:

### CODE REVIEW

File: code/game/machinery/magnet.dm [hunk 1]
```diff
@@ -361,8 +360,7 @@
 		pathpos++ // increase iterator
 
 		// Broadcast the signal
-		spawn()
-			radio_connection.post_signal(src, signal, filter = RADIO_MAGNETS)
+		INVOKE_ASYNC(radio_connection, /datum/radio_frequency/radio_connection.proc/post_signal, src, signal, filter = RADIO_MAGNETS)
 
 		if(speed == 10)
 			sleep(1)
```
Comment (MrStonedOne) [inline code/game/machinery/magnet.dm]: Missed a spot. this sleep is just a delay for the looping = 0 bit below, there is a timer system for setting vars you could use for that instead, avoiding the sleep, or you could make that be a world time that is checked, avoiding the need to use timers or sleeps entirely

File: code/game/machinery/magnet.dm [hunk 2]
```diff
@@ -361,15 +360,14 @@
 		pathpos++ // increase iterator
 
 		// Broadcast the signal
-		spawn()
-			radio_connection.post_signal(src, signal, filter = RADIO_MAGNETS)
+		INVOKE_ASYNC(radio_connection, /datum/radio_frequency.proc/post_signal, src, signal, filter = RADIO_MAGNETS)
```
Comment (Cyberboss) [inline code/game/machinery/magnet.dm]: INVOKE_ASYNC doesn't work with named args

File: code/game/machinery/magnet.dm [hunk 3]
```diff
@@ -361,15 +360,14 @@
...
 
 		if(speed == 10)
-			sleep(1)
+			addtimer(VARSET_CALLBACK(src, looping, FALSE), 1)
 		else
-			sleep(12-speed)
+			addtimer(VARSET_CALLBACK(src, looping, FALSE), 12-speed)
 
-	looping = 0
+	looping = FALSE
```
Comment (Cyberboss) [inline code/game/machinery/magnet.dm]: What you did here doesn't make sense

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Comment by Cyberboss** [comment]
Travis

File: code/game/machinery/magnet.dm [hunk 4]
```diff
@@ -300,8 +300,7 @@
 
 		radio_connection.post_signal(src, signal, filter = RADIO_MAGNETS)
 
-		spawn(1)
-			updateUsrDialog() // pretty sure this increases responsiveness
+		addtimer(CALLBACK(src, .proc/updateUsrDialog()), 1) // pretty sure this increases responsiveness
```
Comment (MrStonedOne) [inline code/game/machinery/magnet.dm]: Remove the `()`

File: code/game/machinery/magnet.dm [hunk 5]
```diff
@@ -361,13 +360,12 @@
...
+		INVOKE_ASYNC(radio_connection, /datum/radio_frequency.proc/post_signal, src, signal, RADIO_MAGNETS)
 
 		if(speed == 10)
-			sleep(1)
+			addtimer(VARSET_CALLBACK(src, looping, FALSE), 1)
 		else
-			sleep(12-speed)
+			addtimer(VARSET_CALLBACK(src, looping, FALSE), 12-speed)
 
 	looping = 0
```
Comment (Cyberboss) [inline code/game/machinery/magnet.dm]: MORTY YOU ADD THE TIMER TO SET IT TO FALSE AND THEN IMMEDIATELY SET IT TO FALSE

File: code/game/machinery/magnet.dm [hunk 6]
```diff
@@ -361,13 +360,12 @@
...
-		spawn()
-			radio_connection.post_signal(src, signal, filter = RADIO_MAGNETS)
+		INVOKE_ASYNC(radio_connection, /datum/radio_frequency.proc/post_signal, src, signal, RADIO_MAGNETS)
 
 		if(speed == 10)
-			sleep(1)
+			addtimer(VARSET_CALLBACK(src, looping, FALSE), 1)
 		else
-			sleep(12-speed)
+			addtimer(VARSET_CALLBACK(src, looping, FALSE), 12-speed)
```
Comment (Cyberboss) [inline code/game/machinery/magnet.dm]: Add a return here, same indent as the if

File: code/game/machinery/magnet.dm [hunk 7]
```diff
@@ -361,13 +360,12 @@
...
```
Comment (Cyberboss) [inline code/game/machinery/magnet.dm]: OK, if you actually read the code, you'd understand why this doesn't preserve the old behaviour at all

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Pull request closed without merging**